src/kudu/consensus/raft_consensus.cc

```diff
@@ -4200,6 +4220,42 @@ Status RaftConsensus::EnableCompressionOnCacheMiss(bool enable) {
   return queue_->log_cache()->EnableCompressionOnCacheMiss(enable);
 }
 
+Status RaftConsensus::SetProxyPolicy(const ProxyPolicy& proxy_policy) {
+  LockGuard l(lock_);
```

Comment:
seems to me the only place where we update routing_table_ is here and in the Init function of RaftConsensus. If we instead had a construct like an atomic shared pointer we could have delegated
all the locking to the Routing tree itself?

Reply:
Discussed offline. Routing table's `UpdateRaftConfig()` is called whenever a new config is proposed/committed/aborted. Hence the topology gets updated there automatically.